src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -175,6 +175,14 @@ public synchronized InAppNotification getNotification(int id, boolean replace) {
         return notif;
     }
 
+    // if a notification was failed to show, add it back to the unseen list so that we
+    // won't lose it
+    public synchronized void markNotificationAsUnseen(InAppNotification notif, boolean isDebugMode) {
+        if (!isDebugMode) {
```

Comment:
If it's debug mode, we add the notification back to the unseen list when `getNotification` gets called. So in this mode, we don't want to add the notification back in this function (as it's already in the unseen list).